expo/expo
# [docs] Upgrade docs to Next 9.5

# Why

This is the "light" version of #10075 and only includes the necessary changes for Next 9.5.

# How

1. Upgraded to Next 9.5+ using the [upgrade guide](https://nextjs.org/docs/upgrading).
2. Moved `static` and `robots.txt` to `public/` to use [new public folder support](https://nextjs.org/blog/next-9-1#public-directory-support).
3. Replaced `/api/` page redirect with `pages/_error` redirect, to [avoid Next handling it as API endpoint](https://nextjs.org/docs/api-routes/introduction).
4. Added workaround for the exported 404-fallback issue when [using `trailingSlash` config](https://github.com/vercel/next.js/issues/16528). → _note: once this is fixed, we can remove the fallback workaround from `package.json`_

# Test Plan

- Run the docs locally with `$ yarn export && yarn export-server` _(validates 1)_
- Check if the robots and static files are returned _(validates 2)_
- Check if the path `/api` still redirects to `/versions/latest` _(validates 3 and 4)_
- Check if both with and without trailing slashes resolves the page on S3
- Run `$ yarn test-links`, while `$ yarn export-server` is running, to validate the redirects
- Run `$ yarn test` to perform the unit tests on components
